### PR TITLE
Add java-imports-scan-local-jars

### PR DESCRIPTION
**Pull request by erjoalgo**
a function to scan the local ~/.m2 repo for class-package associations
addressess #15

### CODE REVIEW

File: java-imports.el [hunk 1]
```diff
@@ -275,6 +275,35 @@ package and cache it for future statements."
...
+	   (jars (shell-command-to-lines
+		  (format "find '%s' -name '*.jar'"
+			  local-repo)))
+	   (cache (pcache-repository java-imports-cache-name)))
+      (dolist (jar jars)
+	(dolist (line (shell-command-to-lines (concat "jar -tf " jar)))
+	  (when (string-match "\\(.*\\)/\\(.*\\)[.]class" line)
+	    (let ((class (intern (match-string 2 line)))
+		  (package-name
+		   (->>
```
Comment (dakrone) [inline java-imports.el]: This adds a dependency on the `dash.el` library, can you add that as a dependency to the file?

File: java-imports.el [hunk 2]
```diff
@@ -275,6 +275,35 @@ package and cache it for future statements."
...
 
+;;;###autoload
+(defun java-imports-scan-local-jars (&optional local-repo)
+  (interactive)
+  "scan the local repository, find local jars and add them to the cache.
+if a class is found in multiple packages, any such package may be registered
+for that class in the cache.
+This is currently a synchronous and potentially slow operation, but
+hopefully faster than adding imports manually or using eclipse"
+  (labels ((shell-command-to-lines (cmd)
```
Comment (dakrone) [inline java-imports.el]: Can you use `cl-labels` instead of `labels`

**Comment by erjoalgo** [comment]
Thanks for reviewing. I totally forgot to remove the -> dash dependency, and I also removed the s.el dependency. I also didn't want to overwrite an existing entry which the user may have manually introduced over an auto-generated one.

**Comment by dakrone** [comment]
Merged this, thanks a ton!

**Comment by erjoalgo** [comment]
Thanks. It looks like replace-regexp-in-string still creates a dependency on the subr-x, while the travis test is using emacs pre-24.4. At the same I added back the s.el dependency. But even if I switched to s-replace, it is merely a proxy for replace-regexp-in-string:

(defun s-replace (old new s)
  "Replaces OLD with NEW in S."
  (replace-regexp-in-string (regexp-quote old) new s t t))

even s-split

(defun s-split (separator s &optional omit-nulls)
  "Split S into substrings bounded by matches for regexp SEPARATOR.
If OMIT-NULLS is non-nil, zero-length substrings are omitted.

This is a simple wrapper around the built-in `split-string'."
  (split-string s separator omit-nulls))

I can't remember the pre-emacs-24.4 way of doing these things, and I'm not sure if it's worth upgrading the emacs version in the test, but I'm sure users of this function would be able to figure it out.

It's interesting that the ac1f24f build passed the tests even though the `replace-regexp-in-string` was already there.

**Pull request merged**